src/map/CMakeLists.txt

```diff
@@ -1,45 +1,125 @@
-#
-# setup
-#
-set( MAP_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}  CACHE INTERNAL "" )
-
-
-#
-# map server
-#
-if( BUILD_SERVERS )
-message( STATUS "Creating target map-server" )
-file(GLOB MAP_HEADERS ${MAP_SOURCE_DIR}/*.hpp)
-file(GLOB MAP_SOURCES ${MAP_SOURCE_DIR}/*.cpp)
-set( DEPENDENCIES common ryml)
-set( LIBRARIES ${GLOBAL_LIBRARIES})
-set( INCLUDE_DIRS ${GLOBAL_INCLUDE_DIRS} ${COMMON_BASE_INCLUDE_DIRS} ${RA_INCLUDE_DIRS} )
-set( DEFINITIONS "${GLOBAL_DEFINITIONS} ${COMMON_BASE_DEFINITIONS}" )
-if( WITH_PCRE )
-	message( STATUS "Enabled PCRE code" )
-	set( LIBRARIES ${LIBRARIES} ${PCRE_LIBRARIES} )
-	set( INCLUDE_DIRS ${INCLUDE_DIRS} ${PCRE_INCLUDE_DIRS} )
-	set( DEFINITIONS "${DEFINITIONS} -DPCRE_SUPPORT" )
-else()
-	message( STATUS "Disabled PCRE code" )
+set(MAP_SOURCES
+	"achievement.cpp"
+	"atcommand.cpp"
+	"battle.cpp"
+	"battleground.cpp"
+	"buyingstore.cpp"
+	"cashshop.cpp"
+	"channel.cpp"
+	"chat.cpp"
+	"chrif.cpp"
+	"clan.cpp"
+	"clif.cpp"
+	"date.cpp"
+	"duel.cpp"
+	"elemental.cpp"
+	"guild.cpp"
+	"homunculus.cpp"
+	"instance.cpp"
+	"intif.cpp"
+	"itemdb.cpp"
+	"log.cpp"
+	"mail.cpp"
+	"map.cpp"
+	"mapreg.cpp"
+	"mercenary.cpp"
+	"mob.cpp"
+	"navi.cpp"
+	"npc.cpp"
+	"npc_chat.cpp"
+	"party.cpp"
+	"path.cpp"
+	"pc.cpp"
+	"pc_groups.cpp"
+	"pet.cpp"
+	"quest.cpp"
+	"script.cpp"
+	"searchstore.cpp"
+	"skill.cpp"
+	"status.cpp"
+	"storage.cpp"
+	"trade.cpp"
+	"unit.cpp"
+	"vending.cpp"
+)
+
+set(MAP_HEADERS
+	"achievement.hpp"
+	"atcommand.hpp"
+	"battleground.hpp"
+	"battle.hpp"
+	"buyingstore.hpp"
+	"cashshop.hpp"
+	"channel.hpp"
+	"chat.hpp"
+	"chrif.hpp"
+	"clan.hpp"
+	"clif.hpp"
+	"clif_obfuscation.hpp"
+	"clif_packetdb.hpp"
+	"clif_shuffle.hpp"
+	"date.hpp"
+	"duel.hpp"
+	"elemental.hpp"
+	"guild.hpp"
+	"homunculus.hpp"
+	"instance.hpp"
+	"intif.hpp"
+	"itemdb.hpp"
+	"log.hpp"
+	"mail.hpp"
+	"map.hpp"
+	"mapreg.hpp"
+	"mercenary.hpp"
+	"mob.hpp"
+	"navi.hpp"
+	"npc.hpp"
+	"packets.hpp"
+	"packets_struct.hpp"
+	"party.hpp"
+	"path.hpp"
+	"pc_groups.hpp"
+	"pc.hpp"
+	"pet.hpp"
+	"quest.hpp"
+	"script_constants.hpp"
+	"script.hpp"
+	"searchstore.hpp"
+	"skill.hpp"
+	"status.hpp"
+	"storage.hpp"
+	"trade.hpp"
+	"unit.hpp"
+	"vending.hpp"
+)
```

Comment:
Consider using the following to automatically introduce all files with hpp suffixes into the MAP_HEADERS variable, so that any future addition of hpp files does not require changes to the CMakeLists.txt file.

`file(GLOB MAP_HEADERS ${MAP_SOURCE_DIR}/*.hpp)`